core/thread/inc/ROOT/TReentrantRWLock.hxx

```diff
@@ -38,8 +38,8 @@ struct UniqueLockRecurseCount {
    using local_t = LocalCounts*;
 
    local_t GetLocal(){
-      TTHREAD_TLS_DECL(LocalCounts, gLocal);
-      return &gLocal;
+       static thread_local LocalCounts unique_local_counts;
```

Comment:
@pcanal can we use `thread_local` throughout all supported platforms these days?

Reply:
Yes, I think so.  The main hold-over was MacOS (and Windows) for objects (i.e. this case).  A priori if the code compile and links, it should be fine.